src/specBuilder/legend/legendUtils.ts

```diff
@@ -222,6 +231,31 @@ const getHighlightOpacityEncoding = (
 		defaultOpacity,
 	];
 };
+//TODO: add comments/tests/etc
+const getHighlightOpAnimationValue = (opacityEncoding: BaseValueRef<number>,): { signal: string } => {
```

Comment:
Using `Op` in the name seems like a code smell that this function isn't doing enough on top of `getHighlightOpacityAnimationValue`.  We could use an import alias to avoid the name collision (OK), or (probably better) collapse the `if ('signal' in opacityEncoding)` logic into `getHighlightOpacityAnimationValue`